src/background/background-firefox.js

```diff
@@ -0,0 +1,33 @@
+const apiUrl = 'https://bff.ecoindex.fr';
+browser.tabs.onUpdated.addListener(async (tabId, changeInfo, tab) => {
+    if (changeInfo.url) {
+        await getAndUpdateEcoindexData(changeInfo.url);
+    }
+});
+
+browser.tabs.onActivated.addListener(async function (activeInfo) {
+    let [tab] = await browser.tabs.query({active: true, currentWindow: true});
+    if (tab.url) {
+        await getAndUpdateEcoindexData(tab.url);
+    }
+});
+
+async function setBadgeUnknownGrade() {
+    await browser.action.setBadgeBackgroundColor({color: '#5b5b5b'});
+    await browser.action.setBadgeText({text: '?'});
+}
+async function getAndUpdateEcoindexData(url) {
+    fetch(`${apiUrl}?url=${url}`)
+        .then((r) => r.json())
+        .then(updateBadge);
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it’s relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->

**❓ question(non-blocking):** Pas de gestion des erreurs ? :)

Reply:
Ne jamais faire confiance... :wink:

Reply:
vu comme ça, c'est plus intéressant d'utiliser setUnknownGrade